src/java.base/unix/native/libnio/fs/UnixNativeDispatcher.c

```diff
@@ -528,14 +678,32 @@ Java_sun_nio_fs_UnixNativeDispatcher_stat0(JNIEnv* env, jclass this,
     int err;
     struct stat64 buf;
     const char* path = (const char*)jlong_to_ptr(pathAddress);
-
+#if defined(__linux__)
+    struct statx statx_buf;
+    int flags = AT_STATX_SYNC_AS_STAT;
+    unsigned int mask = STATX_ALL;
+
+    if (my_statx_func != NULL) {
+        // Prefer statx over stat64 on Linux if it's available
+        RESTARTABLE(statx_wrapper(AT_FDCWD, path, flags, mask, &statx_buf, FOLLOW_SYMLINK), err);
+        if (err == 0) {
+            copy_statx_attributes(env, &statx_buf, attrs);
+            return 0;
+        } else {
+            return errno;
+        }
+    } else {
+#endif
     RESTARTABLE(stat64(path, &buf), err);
     if (err == 0) {
-        prepAttributes(env, &buf, attrs);
+        copy_stat64_attributes(env, &buf, attrs);
         return 0;
     } else {
         return errno;
     }
+#if defined(__linux__)
+    }
+#endif
```

Comment:
Can you replace this with #if defined(__linux) .. #else .. #endif as it will make it easier to read. There are several places where we'll need to fix the indent with nested ifdef usage as it's hard to ready right now.

Reply:
> Can you replace this with #if defined(__linux) .. #else .. #endif as it will make it easier to read.

I'm not sure I follow. We need the `statx` code *and* the old `stat64` code on Linux as that's the fall-back code when `statx` isn't supported. We don't need the `statx` code on other platforms. If I used `#if defined(__linux__) #else ... #endif` instead, I wouldn't get the stat64 code on Linux without duplication. What am I missing?

Reply:
I've added indents. Not sure if it's any better.

Reply:
>  What am I missing?

If you remove the "} else {" then it will be a lot simpler. The thing that I really dislike is the `#if defined(__linux__) } #endif` at the end of each function, it's not needed.

Reply:
Should be fixed.